crates/adapters/src/transport/durable_kafka/input.rs

```diff
@@ -0,0 +1,1041 @@
+}
+
+struct PollerThread {
+    join_handle: Option<JoinHandle<()>>,
+    exit_request: Arc<AtomicBool>,
+}
+
+impl PollerThread {
+    fn new<C: ConsumerContext + 'static>(
+        consumer: &Arc<BaseConsumer<C>>,
+        receiver: &Arc<Mutex<Box<dyn InputConsumer>>>,
+    ) -> Self {
+        let exit_request = Arc::new(AtomicBool::new(false));
+        let join_handle = Some(
+            Builder::new()
+                .name("poller(durable_kafka)".into())
+                .spawn({
+                    let consumer = consumer.clone();
+                    let receiver = receiver.clone();
+                    let exit_request = exit_request.clone();
+                    move || Self::run(consumer, receiver, exit_request)
+                })
+                .unwrap(),
+        );
+        Self {
+            join_handle,
+            exit_request,
+        }
+    }
+
+    fn run<C: ConsumerContext>(
+        consumer: Arc<BaseConsumer<C>>,
+        receiver: Arc<Mutex<Box<dyn InputConsumer>>>,
+        exit_request: Arc<AtomicBool>,
+    ) {
+        while !exit_request.load(AtomicOrdering::Acquire) {
+            match consumer.poll(POLL_TIMEOUT).transpose() {
+                Ok(None) => (),
+                Ok(Some(_)) => {
+                    // All of the subscribed partitions should have be broken
+                    // off into `PartitionQueue`s.
+                    unreachable!()
+                }
+                Err(error) => {
+                    let fatal = error
+                        .rdkafka_error_code()
+                        .is_some_and(|code| code == RDKafkaErrorCode::Fatal);
+                    if !fatal {
+                        receiver.lock().unwrap().error(false, error.into())
+                    } else {
+                        // The client will detect this later.
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl Drop for PollerThread {
+    fn drop(&mut self) {
+        self.exit_request.store(true, AtomicOrdering::Release);
+        let _ = self.join_handle.take().unwrap().join();
+    }
+}
+
+struct WorkerThread {
+    action: Arc<Mutex<Action>>,
+    commit_step: Arc<Mutex<Option<Step>>>,
+    start_step: Step,
+    parker: Parker,
+    config: Arc<Config>,
+    receiver: Arc<Mutex<Box<dyn InputConsumer>>>,
+}
+
+impl WorkerThread {
+    fn spawn(
+        action: &Arc<Mutex<Action>>,
+        commit_step: &Arc<Mutex<Option<Step>>>,
+        start_step: Step,
+        parker: Parker,
+        config: &Arc<Config>,
+        receiver: Box<dyn InputConsumer>,
+    ) -> JoinHandle<()> {
+        let receiver = Arc::new(Mutex::new(receiver));
+        let worker_thread = Self {
+            action: action.clone(),
+            commit_step: commit_step.clone(),
+            start_step,
+            parker,
+            config: config.clone(),
+            receiver: receiver.clone(),
+        };
+        Builder::new()
+            .name("worker(durable_Kafka)".into())
```

Comment:
Thread names are a problem because only the first 15 bytes show up on Linux. Everything after that gets hard-truncated.

I'll use the very short prefix `dkw-`.